### PR TITLE
Remove error from iterator methods

### PR DESCRIPTION
**Pull request by saicheems**

### CODE REVIEW

**Comment by saicheems** [comment]
Resolves #159 

@jba 
@vchudnov-g 
@pongad 
@jmuk

File: src/main/resources/com/google/api/codegen/go/main.snip [hunk 1]
```diff
@@ -227,7 +227,7 @@
                      tokenZeroValue = context.zeroValue(pageStreaming.getRequestTokenField().getType)
 
                     {@methodComment(method, methodName)}
-                    func (client *Client) {@methodName}(ctx context.Context, req {@inTypeName}, opts ...gax.CallOption) (*{@iteratorTypeName}, error) {
+                    func (client *Client) {@methodName}(ctx context.Context, req {@inTypeName}, opts ...gax.CallOption) *{@iteratorTypeName} {
```
Comment (jba) [inline src/main/resources/com/google/api/codegen/go/main.snip]: Not for this PR, perhaps, but I would like to see shorter, more idiomatic variable names:

s/client/c/
s/iterator/it/ (or iter)

**Comment by jba** [comment]
LGTM.

File: src/main/resources/com/google/api/codegen/go/main.snip (hunk 1 above)
Comment (saicheems) [inline src/main/resources/com/google/api/codegen/go/main.snip]: #171

**Pull request merged**